.gitlab-ci.yml

```diff
@@ -0,0 +1,29 @@
+before_script:
+  - docker info
+  # check the availability of some commands:
+  - command >/dev/null -v python3 jq
+
+build-in-docker:
+  stage: build
+  script:
+    - cd tools/docker
+    # Try to pull from dockerhub, build the container images if that fails
+    - ./image-pull.sh || ./image-build.sh
+    # Run clang-format and check there are no modified files by checking git status
+    - ./clang-format.sh
+    - git diff-index --exit-code HEAD
+    - ./build.sh map -f CMAKE_BUILD_TYPE=Debug BUILD_TESTS=ON
+    # Check that the AutoGenerated files are correct by checking git status
+    - git update-index -q --refresh
+    - git diff-index --exit-code HEAD
+  tags:
+    - shell
```

Comment:
Please use the "dockerbuild" tag. If we later switch to one of the other approaches, the tag will stay the same, it will just be a differently-configured runner.